File: services/tradelog/tradelog-api-spec/src/main/java/com/example/tradelog/api/spec/model/ActionType.java
package com.example.tradelog.api.spec.model;

public enum ActionType {
    UNKNOWN,
    STOCK,
    PUT_OPTION,
    CALL_OPTION;

    static public ActionType lookup(String id) {
        try {
            return ActionType.valueOf(id);
        } catch (IllegalArgumentException ex) {
            return UNKNOWN;
        }
    }
}

File: services/tradelog/tradelog-api-spec/src/main/kotlin/com/example/tradelog/api/spec/model/ActionType.kt
package com.example.tradelog.api.spec.model

enum class ActionType {
    UNKNOWN,
    SELL,
    BUY;

    companion object {
        fun lookup(id: String) : ActionType {
            return try {
                valueOf(id)
            } catch (ex: IllegalArgumentException) {
                UNKNOWN
            }
        }
    }
}
